src/passes/OptimizeInstructions.cpp

```diff
@@ -1208,12 +1262,6 @@ struct OptimizeInstructions
             replaceCurrent(curr->left);
             return;
           }
-        } else if (curr->op == Abstract::getBinary(type, Abstract::Sub)) {
-          // we must leave a left zero, as it is how we negate ints
-          if (right && right->value.isZero()) {
-            replaceCurrent(curr->left);
-            return;
-          }
```

Comment:
why was this removed?

Reply:
it's checked `x - 0` which with this canonicalization always should be `x + 0` handled in additional arm

Reply:
Good point, I think you're right. But please add a comment in the code for add,
```
Note that we don't need to handle subtraction: canonicalization avoids a zero on the right anyhow, and a zero on the left can't be removed (it is how we negate ints).
```
Without the comment, it might be confusing why subtraction is not handled.

Reply:
Done